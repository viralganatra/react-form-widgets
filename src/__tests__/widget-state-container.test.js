import React from 'react';
import { shallow, mount } from 'enzyme';
import { shallowToJson, mountToJson } from 'enzyme-to-json';
import WidgetStateContainer from '../widget-state-container';

jest.mock('../events');

const InputWidget = (props) => <input {...props} />;

describe('InputWidget', () => {
    let ComposedComponent;

    beforeEach(() => {
        ComposedComponent = WidgetStateContainer(InputWidget);
    });

    it('should render correctly, passing all props to the decorated component', () => {
        const wrapperFirst = shallow(<ComposedComponent type="text" />);
        const wrapperSecond = shallow(<ComposedComponent type="text" value="test" />);

        expect(shallowToJson(wrapperFirst)).toMatchSnapshot();
        expect(shallowToJson(wrapperSecond)).toMatchSnapshot();
    });

    it('should listen to and call event methods, such as onBlur', () => {
        const methodSpy = jest.fn();
        const wrapper = mount(
            <ComposedComponent
                onKeyPress={methodSpy}
                onKeyUp={methodSpy}
                onKeyDown={methodSpy}
                onBlur={methodSpy}
                onFocus={methodSpy}
                value="foo"
            />,
        );

        expect(mountToJson(wrapper)).toMatchSnapshot();

        wrapper.simulate('change');
        wrapper.simulate('keyPress');
        wrapper.simulate('keyUp');
        wrapper.simulate('keyDown');
        wrapper.simulate('blur');
        wrapper.simulate('focus');

        expect(methodSpy.mock.calls[0][0]).toBe('foo');
        expect(methodSpy.mock.calls[1][0]).toBe('foo');
        expect(methodSpy.mock.calls[2][0]).toBe('foo');
        expect(methodSpy.mock.calls[3][0]).toBe('foo');
    });

    it('should set the value in state when the onChange event occurs and should call the onChange prop with the new value', () => {
        const methodSpy = jest.fn();
        const wrapper = mount(<ComposedComponent onChange={methodSpy} />);

        wrapper.simulate('change', { target: { value: 'testEvent' } });

        expect(mountToJson(wrapper)).toMatchSnapshot();
        expect(methodSpy.mock.calls[0][0]).toBe('testEvent');
    });
});
