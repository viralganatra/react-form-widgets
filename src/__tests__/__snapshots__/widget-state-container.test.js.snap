exports[`InputWidget should listen to and call event methods, such as onBlur 1`] = `
<WidgetStateContainer
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyPress={[Function]}
  onKeyUp={[Function]}
  value="foo">
  <InputWidget
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyPress={[Function]}
    onKeyUp={[Function]}
    value="foo">
    <input
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyPress={[Function]}
      onKeyUp={[Function]}
      value="foo" />
  </InputWidget>
</WidgetStateContainer>
`;

exports[`InputWidget should render correctly, passing all props to the decorated component 1`] = `
<InputWidget
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyPress={[Function]}
  onKeyUp={[Function]}
  type="text"
  value="" />
`;

exports[`InputWidget should render correctly, passing all props to the decorated component 2`] = `
<InputWidget
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyPress={[Function]}
  onKeyUp={[Function]}
  type="text"
  value="test" />
`;

exports[`InputWidget should set the value in state when the onChange event occurs and should call the onChange prop with the new value 1`] = `
<WidgetStateContainer
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyPress={[Function]}
  onKeyUp={[Function]}>
  <InputWidget
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyPress={[Function]}
    onKeyUp={[Function]}
    value="testEvent">
    <input
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyPress={[Function]}
      onKeyUp={[Function]}
      value="testEvent" />
  </InputWidget>
</WidgetStateContainer>
`;
